app/dawn-hub/Views/OnboardingViews/Step2_BlockSettingsView/BlockSettingsView.jsx

```diff
@@ -30,13 +30,70 @@ class BlockSettingsView extends Component {
 		super(props);
 		this.state = {
 			recommendedChoices: false,
-			enable_ad_block: null,
 			kindsOfTrackers: null,
+			enable_ad_block: null,
 			enable_anti_tracking: null,
 			enable_smart_block: null
 		};
 	}
 
+	componentDidMount() {
+		const { setupLifecycle: { blockSetupSeen } } = this.props;
+		if (blockSetupSeen) {
+			const {
+				antiSuite: {
+					enable_anti_tracking,
+					enable_ad_block,
+					enable_smart_block,
+				},
+				blockingPolicy
+			} = this.props;
+
+			const decodedPolicy = this.decodeBlockingPolicy(blockingPolicy);
+
+			this.setState({
+				kindsOfTrackers: decodedPolicy,
+				enable_ad_block,
+				enable_anti_tracking,
+				enable_smart_block
+			});
+		}
+	}
+
+	decodeBlockingPolicy = (blockingPolicy) => {
+		let decodedPolicy;
+		if (typeof blockingPolicy === 'number') {
+			switch (blockingPolicy) {
+				case 1:
+					decodedPolicy = 'BLOCKING_POLICY_EVERYTHING';
+					break;
+				case 2:
+					decodedPolicy = 'BLOCKING_POLICY_RECOMMENDED';
+					break;
+				case 3:
+					decodedPolicy = 'BLOCKING_POLICY_NOTHING';
+					break;
+				default:
+					break;
+			}
+		} else if (typeof blockingPolicy === 'string') {
+			switch (blockingPolicy) {
+				case 'BLOCKING_POLICY_EVERYTHING':
+					decodedPolicy = 1;
+					break;
+				case 'BLOCKING_POLICY_RECOMMENDED':
+					decodedPolicy = 2;
+					break;
+				case 'BLOCKING_POLICY_NOTHING':
+					decodedPolicy = 3;
+					break;
+				default:
+					break;
+			}
+		}
+		return decodedPolicy;
```

Comment:
Also, while we're digging around in this file, could you factor out rendering of the `BlockSettingsView_question` `li`s to a helper like the `renderAnswerBlock` helper I made? It would be a nice bit of cleanup since they are all the same except for a couple strings

Reply:
> Better yet, can we make do without doing these conversions in the first place? I think we are asking for trouble if `kindsOfTrackers` can sometimes be a string and sometimes a number. What if we just always use the string values? That's what `background.js` expects so no conversion would be necessary - only a modification to what values we are testing against in the checkbox rendering code (unless I'm missing somethin'?)

While it looks like we still need the number representation for when we send over the `setup_number` string from onboarding, using the constants everywhere else would mean that we only need to convert to a number for that case in particular, so all the other code can be removed/refactored. Was able to drop quite a few lines with this change.

Reply:
Sweet!